--- a/source/package/network/services/hostapd/src/src/ap/ubus.c
+++ b/source/package/network/services/hostapd/src/src/ap/ubus.c
@@ -157,6 +157,7 @@ hostapd_bss_get_clients(struct ubus_cont
 	struct hostapd_data *hapd = container_of(obj, struct hostapd_data, ubus.obj);
 	struct sta_info *sta;
 	void *list, *c;
+	char *str;
 	char mac_buf[20];
 	static const struct {
 		const char *name;
@@ -175,6 +176,10 @@ hostapd_bss_get_clients(struct ubus_cont
 	};
 
 	blob_buf_init(&b, 0);
+	str = blobmsg_alloc_string_buffer(&b, "ssid", hapd->conf->ssid.ssid_len + 1);
+	memcpy(str, hapd->conf->ssid.ssid, hapd->conf->ssid.ssid_len);
+	str[hapd->conf->ssid.ssid_len] = 0;
+	blobmsg_add_string_buffer(&b);
 	blobmsg_add_macaddr(&b, "bssid", hapd->conf->bssid);
 	blobmsg_add_u32(&b, "freq", hapd->iface->freq);
 	blobmsg_add_u8(&b, "ht_supported", ht_supported(hapd->iface->hw_features));
