--- a/source/package/network/services/hostapd/src/src/ap/ubus.c
+++ b/source/package/network/services/hostapd/src/src/ap/ubus.c
@@ -196,6 +196,20 @@ hostapd_bss_get_clients(struct ubus_cont
 	return 0;
 }
 
+static int
+hostapd_bss_get_chan_utilization(struct ubus_context *ctx, struct ubus_object *obj,
+			struct ubus_request_data *req, const char *method,
+			struct blob_attr *msg)
+{
+	struct hostapd_data *hapd = container_of(obj, struct hostapd_data, ubus.obj);
+	blob_buf_init(&b, 0);
+	blobmsg_add_macaddr(&b, "bssid", hapd->conf->bssid);
+	blobmsg_add_u32(&b, "channel_utilization", hapd->iface->channel_utilization);
+	ubus_send_reply(ctx, req, b.head);
+
+	return 0;
+}
+
 enum {
 	DEL_CLIENT_ADDR,
 	DEL_CLIENT_REASON,
@@ -422,6 +436,7 @@ hostapd_vendor_elements(struct ubus_cont
 
 static const struct ubus_method bss_methods[] = {
 	UBUS_METHOD_NOARG("get_clients", hostapd_bss_get_clients),
+	UBUS_METHOD_NOARG("get_chan_utilization", hostapd_bss_get_chan_utilization),
 	UBUS_METHOD("del_client", hostapd_bss_del_client, del_policy),
 	UBUS_METHOD_NOARG("list_bans", hostapd_bss_list_bans),
 	UBUS_METHOD_NOARG("wps_start", hostapd_bss_wps_start),
